core/src/main/java/de/danoeh/antennapod/core/storage/DBTasks.java

```diff
@@ -388,14 +382,29 @@ public static synchronized Feed updateFeed(Context context, Feed newFeed, boolea
                         savedFeed.getItems().add(idx, item);
                     }
 
-                    // only mark the item new if it was published after or at the same time
-                    // as the most recent item
-                    // (if the most recent date is null then we can assume there are no items
-                    // and this is the first, hence 'new')
-                    // New items that do not have a pubDate set are always marked as new
-                    if (item.getPubDate() == null || priorMostRecentDate == null
+                    // Only send the item to the Inbox if the if the "New Episodes Action"
+                    // preference is set to "Add to Inbox", and one of the following is true:
+                    //
+                    //   1. The item was published after or at the same time as
+                    //      the most recent item
+                    //   2. The most recent date is null (i.e. no previous
+                    //      items and this is the first)
+                    //   3. The item has no pubDate
+                    //
+                    FeedPreferences.NewEpisodesAction feedNewEpisodesAction =
+                            savedFeed.getPreferences().getNewEpisodesAction();
+                    String globalNewEpisodesAction = UserPreferences.getNewEpisodesAction();
+                    boolean shouldAddToInbox = (
+                            feedNewEpisodesAction == FeedPreferences.NewEpisodesAction.ADD_TO_INBOX
+                            ) || (
+                            feedNewEpisodesAction == FeedPreferences.NewEpisodesAction.GLOBAL
+                            && globalNewEpisodesAction.equals(FeedPreferences.NewEpisodesAction.ADD_TO_INBOX.value())
+                            );
```

Comment:
This is quite hard to parse for me. Maybe something like this?

```suggestion
                    FeedPreferences.NewEpisodesAction action = feedNewEpisodesAction;
                    if (action == FeedPreferences.NewEpisodesAction.GLOBAL) {
                        action = UserPreferences.getNewEpisodesAction();
                    }
                    boolean shouldAddToInbox = action == FeedPreferences.NewEpisodesAction.ADD_TO_INBOX;
```